sdk/test/trace/sampler_benchmark.cc

```diff
@@ -0,0 +1,211 @@
+#include "opentelemetry/sdk/trace/tracer.h"
+#include "opentelemetry/sdk/trace/sampler.h"
+#include "opentelemetry/sdk/trace/samplers/always_off.h"
+#include "opentelemetry/sdk/trace/samplers/always_on.h"
+#include "opentelemetry/sdk/trace/samplers/parent_or_else.h"
+#include "opentelemetry/sdk/trace/samplers/probability.h"
+#include "opentelemetry/sdk/trace/simple_processor.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+#include <cstdint>
+
+#include <benchmark/benchmark.h>
+
+using namespace opentelemetry::sdk::trace;
+namespace nostd  = opentelemetry::nostd;
+namespace common = opentelemetry::common;
+using opentelemetry::trace::SpanContext;
+
+/**
+ * A mock sampler that returns non-empty sampling results attributes.
+ */
+class MockSampler final : public Sampler
+{
+public:
+  SamplingResult ShouldSample(const SpanContext * /*parent_context*/,
+                              trace_api::TraceId /*trace_id*/,
+                              nostd::string_view /*name*/,
+                              trace_api::SpanKind /*span_kind*/,
+                              const trace_api::KeyValueIterable & /*attributes*/) noexcept override
+  {
+    // Return two pairs of attributes. These attributes should be added to the span attributes
+    return {Decision::RECORD_AND_SAMPLE,
+            nostd::unique_ptr<const std::map<std::string, opentelemetry::common::AttributeValue>>(
+                new const std::map<std::string, opentelemetry::common::AttributeValue>(
+                    {{"sampling_attr1", 123}, {"sampling_attr2", "string"}}))};
+  }
+
+  std::string GetDescription() const noexcept override { return "MockSampler"; }
+};
+
+/**
+ * A mock exporter that switches a flag once a valid recordable was received.
+ */
+class MockSpanExporter final : public SpanExporter
+{
+public:
+  MockSpanExporter(std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received) noexcept
+      : spans_received_(spans_received)
+  {}
+
+  std::unique_ptr<Recordable> MakeRecordable() noexcept override
+  {
+    return std::unique_ptr<Recordable>(new SpanData);
+  }
+
+  ExportResult Export(const nostd::span<std::unique_ptr<Recordable>> &recordables) noexcept override
+  {
+    for (auto &recordable : recordables)
+    {
+      auto span = std::unique_ptr<SpanData>(static_cast<SpanData *>(recordable.release()));
+      if (span != nullptr)
+      {
+        spans_received_->push_back(std::move(span));
+      }
+    }
+
+    return ExportResult::kSuccess;
+  }
+
+  void Shutdown(std::chrono::microseconds timeout = std::chrono::microseconds(0)) noexcept override
+  {}
+
+private:
+  std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received_;
+};
```

Comment:
Consider pulling this out into a separate file. It might be useful for other tests as well.

Reply:
I took this Mock Exporter from [sdk/test/trace/tracer_test.cc](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/test/trace/tracer_test.cc). I'll create a ticket to move it into a separate file.